test/unit/dom/DocumentFragment.js

```diff
@@ -1,23 +1,59 @@
 /* eslint-env jasmine, mocha */
 
+import create from '../../lib/create';
+
 describe('DocumentFragment', () => {
   it('should report the correct parent node for nested nodes', () => {
     const frag = document.createDocumentFragment();
-    const elem = document.createElement('div');
+    const elem = create();
     frag.appendChild(elem);
     expect(elem.parentNode).to.equal(frag);
   });
 
   it('should remove child elements when appending a document fragment', () => {
-    const div1 = document.createElement('div');
-    const div2 = document.createElement('div');
+    const div1 = create();
+    const div2 = create();
 
     const frag = document.createDocumentFragment();
     frag.appendChild(div1);
     frag.appendChild(div2);
 
-    const elem = document.createElement('div');
+    const elem = create();
     elem.appendChild(frag);
     expect(frag.childNodes.length).to.equal(0);
   });
+
+  it('should correctly report the host node of children that are reparented', () => {
```

Comment:
This is the reproducing test. I used 3 nodes instead of just one to ensure no node is left behind.